handler.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#ifndef SABER_HANDLER_HPP
+#define SABER_HANDLER_HPP
+
+//std
+#include <optional>
+
+namespace saber{
+
+template <typename T>
+class ValueHandler
+{
+public:
+    // ctor stores a copy of the variable's stored data
+    ValueHandler(T& inValue, T inNewValue) :
+        mValue{inValue},
+        mSaved{inValue}
+    {
+        inValue = inNewValue;
+    }
+
+    ~ValueHandler()
+    {
+        Reset();
```

Comment:
Did you know that `~dtor()`s, unlike any other function/method, are always "implicitly `noexcept`"?
https://en.cppreference.com/w/cpp/language/noexcept
This means C++ expects that no exceptions (e.g: `throw`) will occur inside the `~dtor()`.
- If an exception is `throw`n from within a `~dtor()`, the C++ runtime will call `std::terminate()` and kill your App.

Therefore, if you do work inside a `~dtor()`, you: 1) can only call other functions that are declared `noexcept`, or 2) scope any work with a `try/catch (...)` block to prevent the exception from escaping the `~dtor()`.
- Here your `~dtor()` calls `Reset()`... So either `Reset()` must be declared `noexcept`,
or you need `try/catch` around the call to `Reset()`